### PR TITLE
Feat/nsf chart

### PR DESCRIPTION
**Pull request by esherman-credo**
## Describe your changes
Implemented a new credo metric for gain charts.

Returns the number of cumulative samples at each decile, cumulative number of events, and ratio of cumulative events to total events

## Issue ticket number and link
https://credo-ai.atlassian.net/jira/software/c/projects/DSP/boards/29?modal=detail&selectedIssue=DSP-388&assignee=632892e488ed2ebef978635d

## Known outstanding issues that are not fully accounted for
It is not entirely clear whether this is exactly what's been asked for. Multiple blogs point to this implementation as being correct (though we provide fewer columns than many of those). See [here](https://www.listendata.com/2014/08/excel-template-gain-and-lift-charts.html), [here](https://help.salesforce.com/s/articleView?language=en_US&id=sf.bi_edd_model_metrics_cumulative_capture_category.htm&r=https%3A%2F%2Fwww.google.com%2F&type=5), and [here](https://www.geeksforgeeks.org/understanding-gain-chart-and-lift-chart/).

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have built basic tests for new functionality (particularly new evaluators)
- [ ] If new libraries have been added, I have checked that [readthedocs](https://readthedocs.org/projects/credoai-lens/) API documentation is constructed correctly
- [ ] Will this be part of a major product update? If yes, please write one phrase about this update.

## Extra-mile Checklist
- [ ] I have thought expansively about edge cases and written tests for them

### CODE REVIEW

File: credoai/modules/metrics_credoai.py [hunk 1]
```diff
@@ -640,3 +640,30 @@ def skew_parity(items_list, desired_proportions, parity_type="difference"):
...
+        cumulative_ratio[i] = cumulative_events[i] / total_events
+
+    return pd.DataFrame(
+        {
+            "Decile": np.arange(10) + 1,
+            "Cumulative Samples": cumulative_samples,
+            "Cumulative Events": cumulative_events,
+            "Cumulative Gain": cumulative_ratio,
+        }
+    )
```
Comment (IanAtCredo) [inline credoai/modules/metrics_credoai.py:669]:
```suggestion
def credo_gain_chart(y_true, y_prob, bins=10):
    sort_indices = np.argsort(y_prob)[::-1]  # descending order
    total_events = np.sum(y_true)
    overall_rate = np.mean(y_true)

    sorted_y_true = y_true[sort_indices]

    chunked_y_true = np.array_split(sorted_y_true, bins)

    cumulative_samples = np.zeros(bins)
    cumulative_events = np.zeros(bins)
    cumulative_ratio = np.zeros(bins)
    lift = np.zeros(bins)

    for i in range(bins):
        cumulative_samples[i] = len(np.hstack(chunked_y_true[: i + 1]))
        cumulative_events[i] = np.sum(np.hstack(chunked_y_true[: i + 1]))
        cumulative_ratio[i] = cumulative_events[i] / total_events
        lift[i] = (cumulative_events[i] / cumulative_samples[i]) / overall_rate
    return pd.DataFrame(
        {
            "Decile": np.arange(bins) + 1,
            "Cumulative Samples": cumulative_samples,
            "Cumulative Events": cumulative_events,
            "Cumulative Gain": cumulative_ratio,
            "Lift": lift
        }
    )

Added optional bins and lift
```

**Review by IanAtCredo — APPROVED** [review]
Looks great
